src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaptureCollection.cs

```diff
@@ -116,6 +125,120 @@ internal Capture GetCapture(int i)
 
             return _captures[i];
         }
+
+        int IList<Capture>.IndexOf(Capture item)
+        {
+            var comparer = EqualityComparer<Capture>.Default;
+            for (int i = 0; i < Count; i++)
+            {
+                if (comparer.Equals(this[i], item))
+                    return i;
+            }
+            return -1;
+        }
+
+        void IList<Capture>.Insert(int index, Capture item)
+        {
+            throw new NotSupportedException();
+        }
```

Comment:
The reason I didn't include exception text was because I was just following the immutable collections, which don't include any exception text when throwing `NotSupportedException`. If we want the exception text, I can add it. And if so, we should probably open a bug to do the same for the immutable collections.

See https://github.com/dotnet/corefx/blob/dd703af55dc6ae7ad0f40259170324d2ccdf2dad/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList%601.cs#L1151